### PR TITLE
Show dialog before download

### LINKED ISSUE #286: Notify user about total dataset size and download location
Currently, e.g. when downloading VOC (~4GB) for examples/faster_rcnn/ there is no notice how big is the total download and where the file will be stored on disk ($HOME/.chainer/dataset/pfnet/chainercv/hoge). It might be more user-friendly to include such information to help users avoid full hdds.

### PR DESCRIPTION
**Pull request by Hakuyume**
Fix #286 

```
Fetching the size of http://host.robots.ox.ac.uk/pascal/VOC/voc2012/VOCtrainval_11-May-2012.tar ...
File will be saved to /home/admin/.chainer/dataset/_dl_cache/92029705a99338d0932803388148a725.
It will be use 1907.00 MiB of the disk space.
Proceed? (y/N): y
Downloading from http://host.robots.ox.ac.uk/pascal/VOC/voc2012/VOCtrainval_11-May-2012.tar ...
...
```

### CODE REVIEW

**Comment by yuyu2172** [comment]
`request.Request` does not have `method` argument for Python2.

Also, I would be against asking users to confirm download.
The user has already understood that download will happen when necessary.
If unintended downloading starts, the user can just ctrl-c.

File: chainercv/utils/download.py [hunk 1]
```diff
@@ -51,12 +55,12 @@ def cached_download(url):
         str: Path to the downloaded file.
 
     """
-    cache_root = get_dataset_directory('_dl_cache')
+    cache_root = os.path.join(get_dataset_root(), '_dl_cache')
```
Comment (yuyu2172) [inline chainercv/utils/download.py]: Why is this change necessary?

Comment (yuyu2172) [inline chainercv/utils/download.py]: Oh, the following try, except were not necessary.

Comment (Hakuyume) [inline chainercv/utils/download.py]: > Why is this change necessary?

https://github.com/chainer/chainer/pull/2839#issuecomment-308057262

**Comment by Hakuyume** [comment]
> Also, I would be against asking users to confirm download.
The user has already understood that download will happen when necessary.
If unintended downloading starts, the user can just ctrl-c.

We can divide this problem into two.
1. `cache_download` does not tell the size of file and the location where the file will be downloaded. I guess this is the issue reported in #286. I think there is no bad point about showing these information.
2. The download process starts without confirmation. As you suggested, users can stop this by `Ctrl-C`. However, I thought it is more user-freindly to ask explicitly (I'm not sure).

File: chainercv/utils/download.py (hunk 1 above)
Comment (yuyu2172) [inline chainercv/utils/download.py]: I see.

**Comment by yuyu2172** [comment]
> However, I feel it is more user friendly if it asks explicitly

I find it irritating to find my process hang because of this feature.
This is especially the case because this behavior is different from Chainer's download.

**Comment by Hakuyume** [comment]
> I find it irritating to find my process hang because of this feature.
This is especially the case because this behavior is different from Chainer's download.

OK, I removed confirmation step and changed the progress message as follows.
```
Downloading from http://host.robots.ox.ac.uk/pascal/VOC/voc2012/VOCtrainval_11-May-2012.tar to ***/.chainer/dataset/_dl_cache/92029705a99338d0932803388148a725 ...
... 0 %, 0.16 MiB / 1907.00 MiB, 195.06 KiB/s, 0.8 seconds passed
```

**Comment by yuyu2172** [comment]
OK. Thanks.
Perhaps, we can add the remaining time.

**Comment by Hakuyume** [comment]
> Perhaps, we can add the remaining time.

I added.
```
Downloading from http://host.robots.ox.ac.uk/pascal/VOC/voc2012/VOCtrainval_11-May-2012.tar to ***/.chainer/dataset/_dl_cache/92029705a99338d0932803388148a725 ...
... 9 %, 172.07 MiB / 1907.00 MiB, 1145.80 KiB/s, 153.8 seconds passed, 1550.5 seconds remains
```

**Comment by yuyu2172** [comment]
On downloading message, I think it is too verbose.

Four suggestions

+ Decimal numbers are not necessary (224.1MB/s --> 224MB/s)
+ I don't think elapsed time is necessary.
+ How about using "eta <TIME LEFT>" instead of <TIME LEFT> seconds remain. "eta" stands for expected time of arrival, and this is abbreviation used by wget.
+ How about using minutes and seconds instead of seconds for "eta"?

Here is a code I suggest to use from line 32.
```python
    eta = (total_size - progress_size) / speed
    sys.stdout.write(
        '\r... {:.0f}%  {:.0f}MiB/{:.0f}MiB '
        '{:.0f}KiB/s eta {:.0f}m {:.0f}s'
        .format(
            percent, progress_size / (1 << 20), total_size / (1 << 20),
            speed / (1 << 10), eta // 60, eta % 60))
```

**Comment by yuyu2172** [comment]
This is something different from what I have said in the last comment, but how about switching eta to `hours:minutes:seconds`, which can be done with the string below.
Also, I didn't like how outputs length changes from frame to frame.
What do you think about the new output?

```python
    sys.stdout.write(
        '\r... {:3.0f}%  Total {:7.0f}Mib    Current {:7.0f}MiB '
        '{:8.0f}KiB/s    eta {:5.0f}:{:02.0f}:{:02.0f}'    
        .format(
            percent, total_size / (1 << 20), progress_size / (1 << 20),
            speed / (1 << 10), eta // 3600, eta // 60, eta % 60))
```


```...   2%  Total     178Mib    Current       3MiB      570KiB/s    eta     0:05:15
```

**Comment by Hakuyume** [comment]
> I tweaked more, and I found that it looks nice to report only percentage, speed and eta.

From my understanding, the original issue is that users can not know the total dataset size #286. I think it is better to show the total size and downloaded size.

**Comment by Hakuyume** [comment]
> Also, I didn't like how outputs length changes from frame to frame.

I don't  like either. I accept your modification about the length.

**Comment by Hakuyume** [comment]
@yuyu2172 How about this output?
```
Downloading ...
From: http://host.robots.ox.ac.uk/pascal/VOC/voc2007/VOCtest_06-Nov-2007.tar
To: ***/.chainer/dataset/_dl_cache/c320d1dc0cd031efc99e256bf21d57a6
  %   Total    Recv       Speed  Time left
  1  430MiB    5MiB   1634KiB/s    0:04:26
```
This format is inspired by curl command.

**Comment by yuyu2172** [comment]
LGTM
